compiler/src/compile.ml

```diff
@@ -183,10 +183,22 @@ let compile (type reg regx xreg rflag cond asm_op extra_op)
     (fn, f) |> Conv.fdef_of_cufdef |> refresh_i_loc_f |> Conv.cufdef_of_fdef |> snd
   in
 
-  let warning ii msg =
-    (if not !Glob_options.lea then
-     let loc, _ = ii in
-     warning UseLea loc "%a" Printer.pp_warning_msg msg);
+  let warning ii w =
+    let o =
+      match w with
+      | Compiler_util.Use_lea ->
+          if not !Glob_options.lea
+          then Some (UseLea, "LEA instruction is used")
+          else None
+      | Split_memory_access ->
+          let msg =
```

Comment:
Can you add an option like it is done with Glob_options.lea, to say if we print the warning or not ?

Reply:
Would this flag control the warning only or whether we split accesses as well?